### ISSUE TITLE
Feature Request: Minimum stock amount by product group

### ISSUE THREAD

**Issue by reavessm** [grocy/grocy#297]
I'm still relatively new to this, so forgive me if this is already implemented, but I think It would be a nice feature to optionally specify a minimum stock amount by product group.  For example, let's say I have a product group called 'Soda' and two products in that group, namely 'Coke' and 'Mtn Dew'.  I don't care which soda I have really, but I need at least one pack of Soda.  Currently, I can add a userfield to track this, but I can't actually enforce it.

This would be tricky with the 'Add products that are below minimum' button on the shopping list form, but I imagine we could have a drop down and select a product that belongs in that group.

**Comment by berrnd** [comment]
Great idea, we don't have this yet.

Maybe it would be nice to add somehow both/all products of the group to the shopping list, and render it in a way like "4 of any of the following products ...".

One special case to think about would be if the product group and the product itself has a minimum amount defined...

I think this would be also covered by implementing https://github.com/grocy/grocy/issues/196#issuecomment-479939201...

**Comment by reavessm** [comment]
I do like the subgroup idea and that would certainly solve my use-case.  It still might be nice to have it at the product group level if the items aren't quite similar enough to be called a variant, i.e. soda and juice.

**Comment by JoschaMiddendorf** [comment]
That feature would also play nice with recipes where one item can be superseded by another. For example, you should use Onions but red onions or shallots would also work.
In our ERP system at work, we have an exchangeability table in the database for that purpose. When ordering a material, which isn't in stock the database is checked for an exchangeable item.
Maybe that's an idea to start with.
In that way no extra product group has to be created. I like to have my Product groups nicely separated by supermarket sections.

**Comment by berrnd** [comment]
Just to let you know also here: Product variations / sub-products are now read, see https://github.com/grocy/grocy/issues/196#issuecomment-531489697.

This should also work for what @JoschaMiddendorf mentioned above.

But I'll still leave this open, "Minimum stock amount per product group" will come some time... :)

**Comment by blubbel42** [comment]
I love the sub-product feature. still i run into a shopping problem: 
for me its: The Product instant noodles exists, amount 0 in stock. it has 6 flavours of instant noodles as subproducts, with an amount of 1 in stock. I can mark any of these with a minimum amount. then i will be notified that i need to buy cheese-noodles or tomato-noodles or the like. 
But i don't really care which flavour is in stock, as long as three packs are there.
Now, if I mark the instant noodle parent-product with an minimum amount of 3, grocy will tell me to buy 3 packs, even though it correctly says the sum of instant noodle packs is 6, and I really wouldn't need more.
Am I doing something wrong?

**Comment by berrnd** [comment]
> Am I doing something wrong?

No, minimum stock amounts don't consider sub-product amounts currently. I created issue #384 to track this.

This here is still about min. stock amounts per product group.

**Comment by lyz-code** [comment]
@berrnd shouldn't this issue be closed? I think it's already supported

**Comment by berrnd** [comment]
No, currently the min. stock amount can be defined per product, with the option to accumulate all subproduct min. stock amounts in the the parent product (#384). It's still not possible to define a min. stock amount for an **product group**...

**Comment by StefanoGiu** [comment]
I was trying to help here, willing to implement the changes required but I see the main developer is reluctant to offer any help to explain where the changes must be implemented.

I was able to add the min_stock_amount field into the create/edit product group. Now I need to understand how to create table product_groups with the new field min_stock_amount or alter the table in case of existing grocy DB. I'm looking into the SQL inside the migration folder. 

What is the logic that it applies to understand if to run the alter table? When is a new DB created? only when it's not found in the folder?

Thanks for anyone who can help here.

**Comment by StefanoGiu** [comment]
It seems that adding a new sql file into the migration folder it's enough to manage both the new db creation and the alter table for existing DBs.

```
ALTER TABLE product_groups
ADD min_stock_amount INTEGER NOT NULL DEFAULT 0;
```

@berrnd  please kindly confirm

**Comment by berrnd** [comment]
Revisiting this more than 5 years later.

In the meantime the common way of organizing a min. stock amount over multiple products is to use a parent/child product approach + utilizing the product option "Accumulate sub products min. stock amount" which can exactly cover the here mentioned "I don't care which soda I have really" example pretty fine.

**Comment by filipefernandesdev** [comment]
Lets say I have a product group "Meat", and have some products in that group. I want to have at least 2 types of meat at all times. But currently the way to do this is to create a new product and use it as a parent product, which basically makes the "Product group" feature useless.
